[Gerard Jeremiah Matthew] iP

GERARDJM018 opened the pull request:
# **DukePro**

> Stay organized and on top of your tasks effortlessly!

DukePro frees your mind from having to remember things you need to do. it's

- text-based
- easy to learn
- ~~Fast~~ _LIGHTNING_ FAST to use

All you need to do is,

1. download it from [here](https://github.com/GERARDJM018/ip/releases/tag/A-Jar).
2. double click it.
3. add your tasks.
4. let it manage your tasks for you :wink:

And it is **FREE**!!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminder (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

GERARDJM018 committed:
Level 0. Rename, Greet, Exit

GERARDJM018 committed:
Level 1. Echo

GERARDJM018 committed:
Level 2. Add, List

GERARDJM018 committed:
Level 3. Mark as Done

GERARDJM018 committed:
Level 4. ToDos, Events, Deadlines,
Create some changes for the print out method from the previous level

GERARDJM018 committed:
A-TextUiTesting: Automated Text UI Testing
Test: todo, deadline, event, mark, unmark, list, and bye.

GERARDJM018 committed:
Level 5. Handle Errors

GERARDJM018 committed:
Level 6. Delete

GERARDJM018 committed:
A-Enums: Use Enums
Using enumeration for list of commands available

GERARDJM018 committed:
Level-7

Add save file in the directory to save the tasks from the previous session

GERARDJM018 committed:
Merge branch 'branch-Level-7'

GERARDJM018 committed:
Level-8: Add usage of date and times

GERARDJM018 committed:
Merge branch 'branch-level-8'

GERARDJM018 committed:
Implement OOP
Add Command, Ui, Storage, Parser, TaskList Class to implement OOP

GERARDJM018 committed:
Merge branch 'A-MoreOOP'

GERARDJM018 committed:
Implement packaging
put all classes to package and separate them according to the type

GERARDJM018 committed:
Merge branch 'A-Packages'

GERARDJM018 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

GERARDJM018 committed:
Implement gradle

GERARDJM018 committed:
Add JUnit Test
Added TaskListTest, ParserTest, and ToDosTest

GERARDJM018 committed:
Merge branch 'A-JUnit'

GERARDJM018 committed:
A-Jar
Make Jar release compile using java 11

GERARDJM018 committed:
Merge branch 'A-Jar'

GERARDJM018 committed:
Add Javadoc comment to all public method

GERARDJM018 committed:
Merge branch 'master' into A-JUnit

GERARDJM018 committed:
Add JUnit Test
Add JUnit Test for Parser, TaskList, and ToDos

GERARDJM018 committed:
Merge tag 'A-JUnit'

GERARDJM018 committed:
Implement Java Coding Style Standard
Make use of Java coding style standard for every class (some may still be undetected)

GERARDJM018 committed:
Add new command/function
Add FindCommand class to implement find command
Modify Parser, Task, and TaskList to be able to implement find command

GERARDJM018 committed:
Merge branch 'branch-A-JavaDoc'

GERARDJM018 committed:
Apply Java Coding Standards

Ensured adherence to Java coding standards by formatting code, updating naming conventions, and improving code readability. No functional changes were made.

GERARDJM018 committed:
Add FindCommand.java

Made new function for duke to find tasks with specific key.

GERARDJM018 committed:
Add .gitignore

Ignore some .class file which should not be revision controlled

jannaleong commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,55 @@
+
+    /**
+     * Constructor for Duke
+     *
```
Perhaps consider only having 1 line between your method descriptor and the @param line. There are also other places in your code where you have extra lines within your javadoc code. You can consider removing them for neater code ☺️

GERARDJM018 replied:
Did not think about it before, thanks for the suggestion.

jannaleong commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,68 @@
+     * @param no given command.
+     * @return validity of the command.
+     *
+     */
```
Is this comment necessary given that your javadocs already explains the purpose of this method?

GERARDJM018 replied:
Not really, will erase it later. Thanks!

jannaleong commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,68 @@
+     */
+    public static Command parse(String command) throws CommandInvalidException {
+        if (isValid(command)) {
+            String[] inputs = command.split(" ");
```
Are these comments necessary? I believe it is clear what these if else statements in your code are intending to do.

GERARDJM018 replied:
Okay, will delete it soon.

GERARDJM018 replied:
Okay, will delete it soon.

jannaleong commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,79 @@
+     * @throws FileNotFoundException If file is not yet made.
+     */
+    public ArrayList<Task> load() throws FileNotFoundException {
+        ArrayList<Task> taskF = new ArrayList<>();
```
Perhaps change the name is this variable to tasks? Your list contains multiple tasks so its name should be plural.

GERARDJM018 replied:
Oh yea, thanks for noticing it!

jannaleong commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,128 @@
+package duke;
```
Good job for writing clean code, i do not see any violations :)

GERARDJM018 replied:
Thanks!

jannaleong commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,111 @@
+
+    @Override
+    public void execute(TaskList tL, Ui ui, Storage st) {
+        String[] inputs = com.split(" ");
```
Perhaps consider name to names because your array contains multiple names. There are a few other arrays you have in other parts of your code that should have a plural name.

jannaleong commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,111 @@
+                    st.write(tL.getList());
+                }
+                break;
+            case "todo":
```
I believe it would be better if theres a space between if and (. I say this appearing in some other if-else statements as well.

GERARDJM018 replied:
Yeah, I believe that's right too. I tried to do it like that but some of it was still missed. Thanks for noticing!

jannaleong commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,48 @@
+     */
+    public String getCom(){
+        return com;
+    }
```
Consider putting a new line at the end of each of your files.

GERARDJM018 replied:
Okay, thanks for the suggestion!

jannaleong commented on src/main/java/duke/command/UnmarkCommand.java:
```diff
@@ -0,0 +1,55 @@
+    }
+
+    @Override
+    public void execute(TaskList tL, Ui ui, Storage st) {
```
Consider changing the name of this variable to something slightly clearer? Not completely sure what this variable is supposed to do.

GERARDJM018 replied:
Okay, will rename it to be more specific.

jannaleong commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,91 @@
+     * @param st mark status
+     * @param des description for the task.
+     * @param dl deadline date.
+     */
```
Is this how the method should be indented? I believe you are missing 4 spaces.

GERARDJM018 replied:
Will review it later. Thanks

jannaleong commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,73 @@
+     *
+     * @param key the string key need to be contained.
+     * @return a boolean whether the key is contained.
+     */
```
consider renaming this method to is... as it returns a boolean.

GERARDJM018 replied:
Okay, thanks for the suggestion!

jannaleong reviewed (commented):
Overall, it follows the coding standard while but there are just minor sections that could be improved. Good effort!!

GERARDJM018 committed:
Update coding standard

Update coding standard using checkstyle.

GERARDJM018 committed:
Merge branch 'A-CheckStyle'

GERARDJM018 committed:
Add GUI

Add Launcher.java, DialogBox.java, and images to implement GUI. Updated
some codes to implement GUI using javafx

GERARDJM018 committed:
Merge branch 'branch-level-10'

GERARDJM018 committed:
Add MainWindow.java

Implement javafx using FXML to make it easier for designing.

GERARDJM018 committed:
Merge branch 'branch-Level-10'

GERARDJM018 committed:
Update build.gradle

build.gradle use the new main class

The old main class which was on Duke.java is changed to Launcher.java.

The main class is needed to create the new JAR version. Therefore, without changing it
the JAR file cannot run. By changing it, the JAR file can run fine.

GERARDJM018 committed:
Use assertion feature

Use assert feature (not JUnit assertions) to document important assumptions that should hold at various points in the code.

Some variables which supposed to hold some assumption is being checked using assertions.

By checking the assumption, if the assumptions is not hold, some bug can be captured.

GERARDJM018 committed:
Update code quality

Some code or line of code has low readability.

Found some method with more than 30 line of codes and there was also deep nesting in
the code. Fixed it by using SLAP method (refactoring) and extract the deep nesting.

By fixing it, the code readability is enhanced.

GERARDJM018 committed:
Merge pull request #1 from GERARDJM018/branch-A-Assertions

Branch A Assertions

GERARDJM018 committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/duke/Parser.java

GERARDJM018 committed:
Merge pull request #3 from GERARDJM018/branch-A-CodeQuality

Branch A Code Quality

GERARDJM018 committed:
Add duplicate check

The check is needed because the user may add duplicate task to the list.

The program will now check whether the task to be added is already in the list.
If it is in the list, the task will not be added.

The duplicate task will not be added because it has the same details which makes
two same instances of task are redundant and may confuse the user.

GERARDJM018 committed:
Merge branch 'branch-BCD-Extension'

GERARDJM018 committed:
Add UI.png

Add UI.png to creates the product website.
And edit some mistakes regarding naming and code quality.

Add a screenshot of the program as the product website.

To make sure the potential user able to see the GUI.

GERARDJM018 committed:
Edit Ui.png

To make product website the name should be Ui.png.

Changing the name from UI.png to Ui.png.

The naming causes a problem with the product website.

GERARDJM018 committed:
Update README.md

Updates README.md as a user guide.

To make a guide for the new user to understand all available features.

Creates a user guide using GMFD.

GERARDJM018 committed:
Clean the code

Cleans unnecessary print.

GERARDJM018 committed:
Update README.md

Update the screenshot that bugged.
